### DISCUSSION TITLE
Does Quasar work with a Rails backend?

### DISCUSSION THREAD

**Discussion by GandhiGopinath** [quasarframework/quasar#12703] [General - Components / Directives / etc]
Hi: we have a Rails web application that's used by customers. It has one module (page) that we want to move over to Vue/Quasar (reason is performance, better UI components & an eventual goal to deploy it as a mobile app). We are thinking of Vue as a JS framework for its ease of learning & support for incremental migration of pages.

Does Quasar play well with Rails as a back end? Can we just port one part of the application (1 page) to Quasar?

Also, should we start straightaway with Quasar or build the page in Vue and then migrate to Quasar? We are new to both Vue/Quasar and would appreciate any guidance in this regard.

We didn't see too many references to it on the web.

**Comment by customautosys** [comment]
You can use UMD if all you want to port is 1 page.

**Reply by GandhiGopinath** [reply]
Thanks - we will check it out. 

If I later want to port a couple of more pages, can I keep using UMD to incrementally add?

Do I also get the same features/functions if I use UMD, including the ability to build out apps for the pages I have ported?

**Reply by customautosys** [reply]
Yes, you can keep using UMD to incrementally add. No, the features are not exactly the same as Quasar CLI offers a lot more features. If you are using UMD, you are mainly using Quasar as only a library (especially for components) and nothing more.

**Reply by GandhiGopinath** [reply]
So, if I want to spin out an app out of this common codebase (Android or iOS), would I be able to do it if I took the UMD approach? Because that is one of the goals & reasons for considering Quasar

What features of CLI will I miss if I took the UMD approach (at a high level)?

**Reply by customautosys** [reply]
I suppose you could create a separate cordova app and then import the HTML files into it which use the UMD approach.

However, IMHO that would be a very troublesome approach to adopt with practically no reward. It would be far more preferable to use Quasar CLI which will allow you to build for Android / iOS easily using either cordova or capacitor (e.g. quasar build -m android) and you can also build for Electron / Chrome Web Store / Firefox Extension / Server Side Rendering + Progressive Web App at the same time (all of which are not possible with UMD). The downside is that you will then have to create a new project using Quasar CLI and then move your existing code over to it, which I guess would not be an incremental adoption and could take up quite a lot of time.

So far, basically all my Quasar apps (https://customautosys.com/products.html - KetoShare, Sendall, The Viral Game) use the Quasar CLI approach so I have no experience with UMD.

If you're using Rails, I don't think it will work for SSR - SSR requires a node backend (of course you could call the Rails backend using the SSR middleware though).

**Reply by GandhiGopinath** [reply]
Thanks! So, it looks like Quasar CLI is the way to go.

If I wanted to create a Quasar CLI project for just that 1 page (Quasar front end and Rails back end), can it co-exist with the rest of the application (which is built on Ruby on Rails)?

If its possible, what would be the learning curve on Quasar to do this? We have no prior experience in Vue or Quasar, but have Javascript/Ajax experience. This page that I am talking about is a Chat console - there is a decent bit of functionality in it

**Reply by customautosys** [reply]
You can do a quasar CLI project for that 1 page and then when you do an SPA build it will spit out a bundle of webpacked files. So you can maybe put the bundle into a subfolder (maybe https://[path].../chat or something like that) - not sure how to add that to Rails as I don't use Ruby. For communicating with your Rails backend you could use Axios which you can select when you do a "quasar create".

For the chat frontend you can use q-chat-message (this is an example from my app for the comments section on recipes: https://customautosys.com/ketoshare/?recipe=22), It wasn't hard for me to do that - before Quasar I had been using Vue-OnsenUI (Quasar is much better than that).

BTW, it's not necessary to use a node backend if you're not using SSR - I'm currently using PHP as my current webhost only supports PHP (will be switching in June to something that supports node).
